### PR TITLE
DOC-606: Spellchecker deprecation notices

### PR DESCRIPTION
**Pull request by tylerkelly13**
Related Ticket: DOC-606

Description of Changes:
* Add a deprecation notice to 5.4 release notes for the community spellchecker
* Add admonitions to docs about deprecation of the the spellchecker
* Tidy-up some misleading demos
* Fix some language around context menus

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -50,7 +50,8 @@ requires_5_3v: "> **Note**: This feature is only available for TinyMCE 5.3 and l
 requires_5_4v: "> **Note**: This feature is only available for TinyMCE 5.4 and later."
 
 requires_jsscwar_230v: "> **Note**: This feature is only available for TinyMCE self-hosted server-side components, version 2.3.0 and later. To check the version of a running service, visit _&#60;domain&#62;_/_&#60;service&#62;_/version. Such as `http://localhost:8080/ephox-hyperlinking/version`."
-
+deprecate_spellchecker: "> **Important**: The TinyMCE **core** Spell Checker plugin (`spellchecker`) was deprecated with the release of TinyMCE 5.4. For details, see [the TinyMCE core Spell Checker plugin deprecation notice](https://www.tiny.cloud/docs/release-notes/release-notes54/#thetinymcecorespellcheckerplugin)."
```
Comment (lnewson) [inline _config.yml:53]: Dislike `core` here can we please remove it? Core has a very different meaning, as that's just the core editor without any plugins.

File: _includes/configuration/contextmenu_never_use_native.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
 ## `contextmenu_never_use_native`
 
-The `contextmenu_never_use_native` option allows you to disable the browser's native context menu from appearing within the editor.
+The `contextmenu_never_use_native` option allows you to prevent the browser context menu from appearing within the editor.
 
-> Caution: Using this option may result into unexpected behavior when right-clicking in text areas. We advise you to consider all your options carefully before using this feature.
+> **Caution**: Using this option may result into unexpected behavior when right-clicking in text areas. We advise you to consider all your options carefully before using this feature.
```
Comment (lnewson) [inline _includes/configuration/contextmenu_never_use_native.md:5]: `may result into unexpected`? something is wrong here

File: _config.yml (hunk 1 above)
Comment (tylerkelly13) [reply _config.yml:53]: s/core/free/g

File: _includes/configuration/contextmenu_never_use_native.md (hunk 2 above)
Comment (tylerkelly13) [reply _includes/configuration/contextmenu_never_use_native.md]: Fixed by https://github.com/tinymce/tinymce-docs/pull/1600/commits/661608ce08831e40b62b55c36d8e4047b5b710e8

File: _config.yml (hunk 1 above)
Comment (tylerkelly13) [reply _config.yml]: Fixed by https://github.com/tinymce/tinymce-docs/pull/1600/commits/661608ce08831e40b62b55c36d8e4047b5b710e8 and https://github.com/tinymce/tinymce-docs/pull/1600/commits/e6d8bd854f40c0d3ed4da4a222a5d02581680794

**Pull request merged**